packages/react-inbox/src/hooks/use-inbox.ts

```diff
@@ -1,47 +1,144 @@
-import { useCourier } from "@trycourier/react-provider";
+import { useEffect } from 'react';
+import {useCourier} from '@trycourier/react-provider';
+import useTrackEvent from "~/hooks/use-track-event";
+import useMessages from "~/hooks/use-messages";
 
 export default () => {
-  const { dispatch, inbox } = useCourier();
+  const { fetch: fetchMessages } = useMessages();
+  const { dispatch, inbox, transport } = useCourier();
+  const [_, trackEvent] = useTrackEvent();
+
+  const newMessage = (payload) => {
+    dispatch({
+      type: "inbox/NEW_MESSAGE",
+      payload,
+    });
+  };
+
+  const setLoading = (isLoading: boolean) => {
+    dispatch({
+      type: "inbox/SET_LOADING",
+      payload: {
+        isLoading,
+      },
+    });
+  }
+
+  useEffect(() => {
+    transport?.listen({
+      id: "inbox-listener",
+      listener: (courierEvent) => {
+        newMessage(courierEvent?.data);
+      },
+    });
+  }, [transport]);
 
   return {
     ...inbox,
```

Comment:
Would like to see how we can get a more declarative pattern for this, I keep getting confused about whats actually in the `inbox` object. (Don't need to solve this here)

Reply:
i feel the same, right now its basically the main props passed into the component